src/calendar_duration.rs

```diff
@@ -49,6 +49,58 @@ impl fmt::Debug for CalendarDuration {
     }
 }
 
+impl fmt::Display for CalendarDuration {
```

Comment:
I think this should come with a `FromStr` impl and a bunch of round-tripping tests.

Reply:
A parser will be a couple hundred lines. You you want to make it part of this initial PR?

Reply:
Yes, I think so.

Reply:
A complete parser is quite non-trivial :laughing:. It would re-use ca. 75% of #1143.
That is partially because besides the `Pnn̲Ynn̲Mnn̲DTnn̲Hnn̲Mnn̲S` and `Pnn̲W` formats ISO 8601 allows `P<date>T<time>`, with all formats except *week date* supported for those.

Of course I'll start with `Pnn̲Ynn̲Mnn̲DTnn̲Hnn̲Mnn̲S`. That will only share the handling of fractions with #1143.

@djc Can you please review the date and time parsers from that PR, if I split them out into smaller PRs? I know you are busy :disappointed:. Meanwhile I'll continue this parser.

Reply:
I'm confused, how can `P<date>T<time>` be a duration?